WordPressKit/Plugin Management/SelfHostedPluginManagementClient.swift

```diff
@@ -7,89 +7,74 @@ public class SelfHostedPluginManagementClient: PluginManagementClient {
 
     // MARK: - Get
     public func getPlugins(success: @escaping (SitePlugins) -> Void, failure: @escaping (Error) -> Void) {
-        remote.GET(path(), parameters: nil) { (result, _) in
-            switch result {
-                case .success(let responseObject):
-                    guard let response = responseObject as? [[String: AnyObject]] else {
-                        failure(PluginServiceRemote.ResponseError.decodingFailure)
-                        return
+        Task { @MainActor in
+            await remote.get(path: path(), type: [PluginStateResponse].self)
+                .mapError { error -> Error in
+                    if case let .unparsableResponse(_, _, underlyingError) = error, underlyingError is DecodingError {
+                        return PluginServiceRemote.ResponseError.decodingFailure
                     }
+                    return error
+                }
+                .map {
+                    SitePlugins(
+                        plugins: $0.compactMap { self.pluginState(with: $0) },
+                        capabilities: SitePluginCapabilities(modify: true, autoupdate: false)
+                    )
+                }
+                .execute(onSuccess: success, onFailure: failure)
 
-                    let plugins = response.compactMap { (obj) -> PluginState? in
-                        self.pluginState(with: obj)
-                    }
-
-                    let result = SitePlugins(plugins: plugins,
-                                             capabilities: SitePluginCapabilities(modify: true, autoupdate: false))
-                    success(result)
-
-                case .failure(let error):
-                    failure(error)
-            }
         }
     }
 
     // MARK: - Activate / Deactivate
     public func activatePlugin(pluginID: String, success: @escaping () -> Void, failure: @escaping (Error) -> Void) {
-        let parameters = ["status": "active"] as [String: AnyObject]
+        let parameters = ["status": "active"]
         let path = self.path(with: pluginID)
-        remote.request(method: .put, path: path, parameters: parameters) { (result, _) in
-            switch result {
-                case .success:
-                    success()
-
-                case .failure(let error):
-                    failure(error)
-            }
+        Task { @MainActor in
+            await remote.perform(.put, path: path, parameters: parameters, type: AnyResponse.self)
```

Comment:
Have you considered adding methods for `PUT` and other HTTP verbs, for consistency with the `POST` and `GET` implementations?

Reply:
Also, the `AnyResponse` type and related `.map { _ in }` took me a bit to understand.

I read them as a way to deal with an API which returns no data, or for which we don't care about the data.

What do you think of pushing this knowledge down at the `WordPressOrgRestApi` level?

We could have a `perform` implementation that returns `WordPressAPIResult<Void, WordPressOrgRestApiError>`, allowing us to segregate the `AnyResponse + .map { _ in }` "trick" lower into the implementation layer.

If instead it was implemented at this level because it's a one-off kind of thing, I'd still suggest centralizing the implementation into a `fileprivate extension WordPressOrgRestApi`

Reply:
> Have you considered adding methods for PUT and other HTTP verbs, for consistency with the POST and GET implementations?

The post and get functions are convenience functions because the majority of HTTP requests uses these two methods. I didn't add convenience functions for other methods simply because they are not used by many places (maybe only one or two places?).

> What do you think of pushing this knowledge down at the WordPressOrgRestApi level?

IMO, It's kind of strange to ignore the response. Adding such support to `WordPressOrgRestApi` feels like we are encouraging it, but I don't think we should... Also, I believe this is the only place that ignores HTTP API response. I don't want to complicate `WordPressOrgRestApi` too much for this one case.